### DISCUSSION TITLE
capcha resolvers api

### DISCUSSION THREAD

**Discussion by KOLBASOID** [jef/streetmerchant#2513] [Ideas and feature requests]
Please add the ability to send captchas to the service https://2captcha.com/2captcha-api   for an automatic solution

**Comment by ArmaanCPro** [comment]
I hope they do. Have you found any solution to manually handle it? I saw another thread in October of 2020 requesting the same thing. There is also this link for @jef to check out: https://github.com/jef/streetmerchant/discussions/2426

**Comment by jef** [comment]
We currently have a free alternative to this and you can read more about it at https://www.jef.buzz/streetmerchant/reference/captcha/.

I also wouldn't mind implementing something like this for folks that want to pay for some of this automation.

CC: @klords, you have any thoughts?

**Reply by klords** [reply]
Yeah, it's not a bad idea since our current handler only works on image-based captchas, and not on things like recaptcha/hcaptcha. I know both puppeteer-extra and playwright-extra provide a plugin for recaptcha/hcaptcha that integrates with this service. We could incorporate it with our captcha handler (or just use the underlying library directly) and let users opt-in to it, since a few people seem to want this integration. This also doesn't preclude anyone motivated enough from enhancing the current handler to include support for these captcha mechanisms.

Also on this topic, I've also looked at some of the automatic captcha solvers for Amazon and the options there are to invoke a python subroutine to make use of existing libraries or adapt one to run using JS libs (or roll our own entirely). The current captcha handler would still serve as a fallback in these cases.

**Comment by jiglesiasc** [comment]
I vote for this. Zotac Store use hCaptcha and is really annoying
